app_dart/lib/src/request_handlers/check_flaky_tests_and_update_github.dart

```diff
@@ -0,0 +1,450 @@
+        'Statuses': e.toString(),
+        'stack': stack.toString(),
+      });
+    }
+    return Body.forJson(const <String, dynamic>{
+      'Statuses': 'success',
+    });
+  }
+
+  double get _threshold => double.parse(request.uri.queryParameters[_thresholdKey]);
+
+  String _generateNewRef() {
+    const String chars = 'AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz1234567890';
+    final Random rnd = Random();
+    final String randomString = String.fromCharCodes(Iterable<int>.generate(
+        10, (_) => chars.codeUnitAt(rnd.nextInt(chars.length))));
+    return '${_refsPrefix}marks-flaky-$randomString';
+  }
+
+  Future<Map<String, BuilderStats>> _getBuilderStats(GitHub client) async {
+    const String projectId = 'flutter-dashboard';
+
+    final JobsResourceApi jobsResourceApi = await config.createJobsResourceApi();
+    final QueryRequest query = QueryRequest.fromJson(<String, Object>{
+      'query': getFlakyRateQuery,
+      'useLegacySql': false
+    });
+    final QueryResponse response = await jobsResourceApi.query(query, projectId);
+    if (!response.jobComplete) {
+      throw 'job does not complete';
+    }
+    final Map<String, BuilderStats> nameToStats = <String, BuilderStats>{};
+    final String testOwnerContent = await _getTestOwnerContent(client);
+    final YamlMap ci = loadYaml(await _getCIContent(client)) as YamlMap;
+    final YamlList targets = ci[_ciYamlTargetsKey] as YamlList;
+    for (final TableRow row in response.rows) {
+      final String builder = row.f[0].v as String;
+      List<String> failedBuilds = (row.f[3].v as String)?.split(', ');
+      failedBuilds?.sort();
+      failedBuilds = failedBuilds?.reversed?.toList();
+      List<String> succeededBuilds = (row.f[4].v as String)?.split(', ');
+      succeededBuilds?.sort();
+      succeededBuilds = succeededBuilds?.reversed?.toList();
+      final YamlMap target = targets.firstWhere(
+        (dynamic element) => element[_ciYamlTargetBuilderKey] == builder,
+        orElse: () => null,
+      ) as YamlMap;
+      if (target == null) {
+        continue;
+      }
+      final String testName = target[_ciYamlTargetNameKey] as String;
+      nameToStats[builder] = BuilderStats(
+        name: builder,
+        flakyRate: double.parse(row.f[7].v as String),
+        failedBuilds: failedBuilds ?? const <String>[],
+        succeededBuilds: succeededBuilds ?? const <String>[],
+        recentCommit: row.f[5].v as String,
+        failedBuildOfRecentCommit: row.f[6].v as String,
+        testOwner: await _findTestOwner(testName, testOwnerContent)
+      );
+    }
+    return nameToStats;
+  }
+
+  Future<Map<String, ExistingGithubIssue>> _getExistingGithubIssues(GitHub client) async {
+    final Map<String, ExistingGithubIssue> nameToExistingIssue = <String, ExistingGithubIssue>{};
+    await for (final Issue issue in client.issues.listByRepo(
+        config.flutterSlug, state:'all', labels: <String>[kFlakeLabel])) {
```

Comment:
shall we use the `severe: flake` label to limit the search range? `team: flake` contains lots of other flaky issues that we may not be interested in.